modules/integration/tests-common/integration-tests-utils/src/main/java/org/wso2/mb/integration/common/utils/ui/pages/main/HomePage.java

```diff
@@ -34,39 +34,104 @@
 
     /**
      * Checks whether the current page is the home page. if not throws a runtime exception
+     *
      * @param driver WebDriver
      */
     public HomePage(WebDriver driver) {
         super(driver);
         // Check that we're on the right page.
-        if (!driver.findElement(By.id(UIElementMapper.getInstance()
-                .getElement("home.dashboard.middle.text"))).getText().contains("Home")) {
+        if (!driver.findElement(By.id(
+                UIElementMapper.getInstance().getElement("home.dashboard.middle.text"))).getText()
+                                                                                .contains("Home")) {
             throw new IllegalStateException("This is not the home page");
         }
     }
 
-    public DLCBrowsePage getDLCBrowsePage() throws Exception {
-        driver.findElement(By.xpath(UIElementMapper.getInstance().getElement("home.mb.dlc.browse.xpath"))).click();
+    /**
+     * Clicks on the dead letter channel browsing page.
+     *
+     * @return A {@link org.wso2.mb.integration.common.utils.ui.pages.main.DLCBrowsePage}
+     * @throws IOException
+     */
+    public DLCBrowsePage getDLCBrowsePage() throws IOException {
+        this.clickOnMenuItem("home.mb.dlc.browse.xpath");
         return new DLCBrowsePage(driver);
     }
 
+    /**
+     * Clicks on the queue browsing page.
+     *
+     * @return A {@link org.wso2.mb.integration.common.utils.ui.pages.main.QueuesBrowsePage}
+     * @throws IOException
+     */
     public QueuesBrowsePage getQueuesBrowsePage() throws IOException {
-        driver.findElement(By.xpath(UIElementMapper.getInstance().getElement("home.mb.queues.browse.xpath"))).click();
+        this.clickOnMenuItem("home.mb.queues.browse.xpath");
         return new QueuesBrowsePage(driver);
     }
 
-    public TopicBrowsePage getTopicBrowsePage() throws IOException {
-        driver.findElement(By.xpath(UIElementMapper.getInstance().getElement("home.mb.topic.browse.xpath"))).click();
-        return new TopicBrowsePage(driver);
-    }
-
+    /**
+     * Clicks on the queue adding page.
+     *
+     * @return A {@link org.wso2.mb.integration.common.utils.ui.pages.main.QueueAddPage}
+     * @throws IOException
+     */
     public QueueAddPage getQueueAddPage() throws IOException {
-        driver.findElement(By.xpath(UIElementMapper.getInstance().getElement("home.mb.queues.add.xpath"))).click();
+        this.clickOnMenuItem("home.mb.queues.add.xpath");
```

Comment:
This resource is only being used in this class so far. Therefore keeping this as is.